mockkubeapiserver/schema.go

```diff
@@ -0,0 +1,89 @@
+package mockkubeapiserver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v4/merge"
+	"sigs.k8s.io/structured-merge-diff/v4/typed"
+
+	"sigs.k8s.io/kubebuilder-declarative-pattern/mockkubeapiserver/forked"
+)
+
+type mockSchema struct {
+	builtin   *Schema
+	resources []*ResourceInfo
+}
+
+func (s *mockSchema) Init() error {
+	schema, err := KubernetesBuiltInSchema()
+	if err != nil {
+		return err
+	}
+	s.builtin = schema
+	return nil
+}
+
+type typeInfo struct {
+	ParserType typed.ParseableType
+}
+
+func (r *ResourceInfo) DoServerSideApply(ctx context.Context, live *unstructured.Unstructured, patchYAML []byte, options metav1.PatchOptions) (*unstructured.Unstructured, bool, error) {
+	if r.TypeInfo == nil {
+		return nil, false, fmt.Errorf("no type info for %v", r.GVK)
+	}
+
+	updater := merge.Updater{}
+
+	liveObject, err := r.TypeInfo.ParserType.FromUnstructured(live.Object)
+	if err != nil {
+		return nil, false, fmt.Errorf("error parsing live object: %w", err)
+	}
+
+	configObject, err := r.TypeInfo.ParserType.FromYAML(typed.YAMLObject(patchYAML))
+	if err != nil {
+		return nil, false, fmt.Errorf("error parsing patch object: %w", err)
+	}
+	force := false
+	if options.Force != nil {
+		force = *options.Force
+	}
+	var managers fieldpath.ManagedFields
+	manager := metav1.ManagedFieldsEntry{
+		Manager: options.FieldManager,
+	}
+	// TODO: This is surprising ... the manager key is not the manager key, but rather the json-encoded form of the object
+	// (at least Decode assumes this)
```

Comment:
Yeah, I believe you it's surprising! :D
you should add probably the operation here so that we don't mix the two (they don't share the same names)